### DISCUSSION TITLE
Could NiceGUI build a board game gui via PLOT instead of PNG/SVG (issue#869)? And is it compatible with Termux?

### DISCUSSION THREAD

**Discussion by HackYardo** [zauberzeug/nicegui#3479] [Q&A] [answered]
### Question

It seems that no one has mentioned **Termux**, a Linux emulator on Android. One can do many c++, py or js projects on Android with the emulator. As NiceGUI is a web-ui python library, how about the compatibilities that works with *Termux*?

I want to make a board game "web-ui based on plots and widgets", or a "dashboard" or a "notebook" in some terms. In details, chess for example:

- users hover over a plot (users are thinking)
- users click on the plot (users move pieces)
- the plot updates
- python scripts ask *Stockfish* or *LeelaChessZero* to move
- the plot updates
- waiting for users' next move

 Is NiceGUI a proper library to implement a board game web-ui based on plots and widgets?

**Comment by falkoschindler** [comment, marked as answer]
Creating a board game with NiceGUI: Sure, this should work.

Regarding NiceGUI on mobile: See discussions #3423, #1374 and #578. Termux hasn't been mentioned yet, so maybe that's worth to investigate.

**Comment by HackYardo** [comment]
@falkoschindler I tried to run `repo/main.py` yesterday, **it works fine** on my ARM64 phone, just do what you do as on Linux AMD64, here are the main steps:
```
# download Termux
# download a Linux distro for Termux
# enter the distro, e.g. MFDGaming/ubuntu-in-termux
apt update
apt install python pip
pip install nicegui pytest selenium isort --timeout=300
wget link/to/repo.zip
unzip repo.zip
cd repo/
python3 main.py
# open a broswer, e.g. Firefox
# enter localhost:8080
``` 

However, sorry to bother you again, I only know a bit python, is there a mouseEvent-plotScatterDraw demo of NiceGUI? And which plot lib you would recommand? *Matplotlib, Plotly, Echart or HighCharts?*

**Reply by falkoschindler** [reply]
> is there a mouseEvent-plotScatterDraw demo

Sorry, I don't know what you mean.

**Reply by rodja** [reply]
I wonder if a plot library is the right choice at all. Why not use an `ui.interactive_image` wir empty canvas and SVG content to draw the game visuals?

**Reply by HackYardo** [reply]
@falkoschindler It means **"using plotEvent to update a plot, not widgetEvent"**, and there is a method `nicegui.ui.plotlib.on('eventName', 'eventHandler')`.

**Reply by HackYardo** [reply]
@rodja The cause is modern board game bots could **not only play but also review/analyze games**. And analyze will generate much data. You will need a plot lib to visualize so much data.

**Reply by rodja** [reply]
Sure. The data plots should be drawn with plots. But the game GUI itself can be something different.
